wavefront/server/packages/flo_cloud/flo_cloud/postgres/postgres.py

```
@@ -0,0 +1,261 @@
+            )
+            processed_join = processed_join.replace(f'{table_name}.', f'{alias}.')
+            processed_where = processed_where.replace(f'{table_name}.', f'{alias}.')
+
+        group_by_clause = f'GROUP BY {group_by}' if group_by else ''
+        order_by_clause = f'ORDER BY {order_by}' if order_by else ''
+        base_table = f'{table_prefix}{table_names[0]}'
+        return (
+            f'SELECT {projection} FROM {base_table} AS {aliases[0]} '
+            f'{processed_join} WHERE {processed_where} '
+            f'{group_by_clause} {order_by_clause} '
+            f'LIMIT {limit} OFFSET {offset}'
```

Ruff (0.15.10)

[error] Possible SQL injection vector through string-based query construction

(S608)